03-transactions.md

```diff
@@ -117,11 +117,36 @@ If a revoked commitment transaction is published, the other party can spend this
 
 #### `to_remote` Output
 
-This output sends funds to the other peer and thus is a simple P2WPKH to `remotepubkey`.
+This output sends funds to the other peer, thus is not encumbered by a
+revocation private key.
+
+If `option_anchor_outputs` applies to the commitment transaction, the `to_remote` output is delayed similarly to the `to_local` output, and is to a fixed key:
+
+    `remote_to_self_delay`
+    OP_CSV
+    OP_DROP
+    <remote_pubkey>
+    OP_CHECKSIG
+
+The output is spent by a transaction with `nSequence` field set to `to_remote_self_delay` (which can only be valid after that duration has passed) and witness:
+
+    <remote_sig>
+
+Otherwise, this output is a simple P2WPKH to `remotepubkey`.
+
+
+#### `to_local_anchor` and `to_remote_anchor` Output (option_anchor_outputs)
+
+This output can be spent by the local and remote nodes respectively to provide incentive to mine the transaction, using child-pays-for-parent. For simplicity, both
+anchor outputs are always added. The only case where this adds some unnecessary weight is if the initiator closes right after opening (no `to_local` output). The
```

Comment:
Isn't the `to_remote` output which is absent in case of under the dust limit `push_msat` ? Also is that hard to trim the anchor output if the party doesn't have a balance? I think it may worth it to avoid utxo set pollution..

Reply:
Ok good point that there can also be no `to_remote` if the balance is above zero but below dust. But in that case, there may still be pending htlcs for which the remote wants to anchor down the commit tx. If those are not present either, we could trim the the anchor indeed.

Reply:
Updated PR to only produce an anchor if there are htlcs and/or the output is non dust.